docs/reference/esql/esql-multi-index.asciidoc

```diff
@@ -0,0 +1,180 @@
+
+Use the format `<remote_cluster_name>:<target>` to query data streams and indices
+on remote clusters:
+
+[source,esql]
+----
+FROM cluster_one:employees-00001,cluster_two:other-employees-*
+----
+
+See <<esql-cross-clusters, using {esql} across clusters>>.
+
+[discrete]
+[[esql-multi-index-invalid-mapping]]
+=== Invalid field mappings
+
+When querying multiple indexes, data streams or aliases, it is possible that the same field is mapped to multiple different types.
+For example, consider the two indexes with the following field mappings:
+
+*index: events_ip*
+```
+{
+  "mappings": {
+    "properties": {
+      "@timestamp":     { "type": "date" },
+      "client_ip":      { "type": "ip" },
+      "event_duration": { "type": "long" },
+      "message":        { "type": "keyword" }
+    }
+  }
+}
+```
+
+*index: events_keyword*
+```
+{
+  "mappings": {
+    "properties": {
+      "@timestamp":     { "type": "date" },
+      "client_ip":      { "type": "keyword" },
+      "event_duration": { "type": "long" },
+      "message":        { "type": "keyword" }
+    }
+  }
+}
+```
+
+When querying each of these individually, with a query as simple as `FROM events_ip`, the results will be provided with type specific columns:
+
+[source.merge.styled,esql]
+----
+FROM events_ip
+| SORT @timestamp DESC
+----
+[%header.monospaced.styled,format=dsv,separator=|]
+|===
+@timestamp:date | client_ip:ip | event_duration:long | message:keyword
+2023-10-23T13:55:01.543Z | 172.21.3.15  | 1756467 | Connected to 10.1.0.1
+2023-10-23T13:53:55.832Z | 172.21.3.15  | 5033755 | Connection error
+2023-10-23T13:52:55.015Z | 172.21.3.15  | 8268153 | Connection error
+|===
+
+Note how the `client_ip` column is correctly identified as of type `ip`, and all values are displayed.
+However, if instead the query sources two conflicting indexes with `FROM events_*`, the type of the `client_ip` column cannot be determined,
+and will be reported as `unsupported` with all values returned as `null`.
+
+[source.merge.styled,esql]
+----
+FROM events_*
+| SORT @timestamp DESC
+----
+[%header.monospaced.styled,format=dsv,separator=|]
+|===
+@timestamp:date | client_ip:unsupported | event_duration:long | message:keyword
+2023-10-23T13:55:01.543Z | null  | 1756467 | Connected to 10.1.0.1
+2023-10-23T13:53:55.832Z | null  | 5033755 | Connection error
+2023-10-23T13:52:55.015Z | null  | 8268153 | Connection error
+2023-10-23T13:51:54.732Z | null  | 725448  | Connection error
+2023-10-23T13:33:34.937Z | null  | 1232382 | Disconnected
+2023-10-23T12:27:28.948Z | null  | 2764889 | Connected to 10.1.0.2
+2023-10-23T12:15:03.360Z | null  | 3450233 | Connected to 10.1.0.3
+|===
+
+In addition, if the query refers to this unsupported field directly, the query will fail:
+
+[source.merge.styled,esql]
+----
+FROM events_*
+| KEEP @timestamp, client_ip, event_duration, message
+| SORT @timestamp DESC
+----
+
+....
+Cannot use field [client_ip] due to ambiguities being mapped as [2] incompatible types:
```

Comment:
It would be nice to have a better styling for this error message. Right now this style does not really stand out sufficiently. Do we have a convention for displaying errors? A quick search did not find anything.

Reply:
I'm still looking into options and conventions, but for now I changed this to use `[source,bash]`, which is kind of a default and looks decent ([example](https://www.elastic.co/guide/en/app-search/current/troubleshooting.html?edit#troubleshooting-java-error)). WDYT?

Reply:
Certainly much better than my approach, although not yet quite perfect. Good enough for now.